packages/utils/recursive-parser.js

```diff
@@ -4,38 +4,51 @@ const utils = require("./ast-utils");
 
 module.exports = function recursiveTransform(j, ast, value, action, key) {
 	const node = utils.findRootNodesByName(j, ast, key);
+
+	// get module.exports prop
+	const root = ast
+		.find(j.ObjectExpression)
+		.filter(p => {
+			return (
+				utils.safeTraverse(p, [
+					"parentPath",
+					"value",
+					"left",
+					"object",
+					"name"
+				]) === "module" &&
+				utils.safeTraverse(p, [
+					"parentPath",
+					"value",
+					"left",
+					"property",
+					"name"
+				]) === "exports"
+			);
+		})
+		.filter(p => p.value.properties);
+
 	if (node.size() !== 0) {
-		// push to existing key
-		return ast;
+		// select node with existing key
+		return utils.findRootNodesByName(
+			j, root, key
+		)
+			.forEach(p => {
+				if (action === "add") {
+				// update property/replace
+					j(p).replaceWith(
+						utils.updateProperty(j, p, key, value)
+					);
+				}
+			});
 	} else {
-		// get module.exports prop
-		const root = ast
-			.find(j.ObjectExpression)
-			.filter(p => {
-				return (
-					utils.safeTraverse(p, [
-						"parentPath",
-						"value",
-						"left",
-						"object",
-						"name"
-					]) === "module" &&
-					utils.safeTraverse(p, [
-						"parentPath",
-						"value",
-						"left",
-						"property",
-						"name"
-					]) === "exports"
-				);
-			})
-			.filter(p => p.value.properties);
 		return root.forEach(p => {
 			if (key === "topScope") {
 				utils.parseTopScope(j, p, value);
 			} else if (key === "merge") {
 				utils.parseMerge(j, p, value);
 			} else {
+				// init, add new property
```

Comment:
comment right after conditional if you're gonna comment it. Also, I think it's clear what we are doing here

Reply:
This part would be used in _add_ command as well in cases when a property doesn't exist. For instance, when someone tries to add `mode` property which isn't present in config, `node.size()` will be 0 and this `addProperty()` would be called. That's why want to make super clear that this part of the code is getting used in _init_ and _add_.
In cases when `mode` is already present and we're overwriting it, `updateProperty()` will be called. Basically, _add_ and _update_ commands would work same.

Reply:
Maybe we can just have three options: `add` (renamed from init), `update` and `remove`